Source/dvlnet/tcp_client.cpp

```diff
@@ -28,10 +29,40 @@ int tcp_client::join(std::string addrstr)
 	constexpr int MsSleep = 10;
 	constexpr int NoSleep = 250;
 
-	std::string port = StrCat(*sgOptions.Network.port);
+	const char *defaultPort = "6112";
+	std::string_view host = "";
+	std::string_view port = defaultPort;
+	if (addrstr[0] == '[') {
+		// Assume IPv6 address in square brackets, followed by port
+		// Example: [::1]:6113
+		auto it = std::find(addrstr.begin() + 1, addrstr.end(), ']');

```

Comment:
Using positions rather than iterators will likely result in somewhat easier to read code here

```suggestion
		size_t pos = addrstr.find(']', 1);
```